inst/eDISH_app/server.R

```diff
@@ -0,0 +1,204 @@
+    
+    updateRadioButtons(session, "select_file", "Select file for eDISH chart",
+                              choices = names(dd$data))
+    
+  })
+
+
+  # get selected dataset when selection changes
+  data_selected <- eventReactive(input$select_file, {
+    if (! input$select_file == "No files available"){
+      isolate({index <- which(names(dd$data)==input$select_file)[1]})
+      dd$data[[index]]
+    } else{
+      return()
+    }
+    })
+
+ # upon a dataset being uploaded and selected, generate data preview
+  output$data_preview <- DT::renderDataTable({
+      DT::datatable(data = data_selected(),
+                     rownames = FALSE,
+                     style="bootstrap",
+                     class="compact",
+                      extensions = "Scroller", options = list(scrollY=400, scrollX=TRUE))
+  })
+
+
+  # upon a dataset being selected, run detectStandard() function
+  standard <- reactive({
+     req(data_selected())
+     detectStandard(data_selected())$standard
+  })
+
+
+  # output UI message based on detectStandard() result
+   output$detectStandard_msg <- renderUI({
+     req(standard())
+     if (standard()=="None"){
+       HTML("No standard detected. Please use settings tab to configure chart.")
+     } else {
+       HTML(paste("Matched",standard(),"data standard. eDISH chart available."))
+     }
+  })
+
+  # upon a dataset being selected, use generateSettings() to produce a settings obj
+  settings_list <- reactiveValues(settings = NULL)
+  observe({
+    req(data_selected())
+    standard()
+    settings_list$settings <- generateSettings(standard = standard(), chart = "eDish")
+  }) 
+  
+
+  # run validateSettings(data, standard, settings) and return a status
+  # this bombs if standard="none" because we are not yet updating settings obj based on
+  #   user input. Once we allow that, run validateSettings under either:
+  #     (1) standard is AdAM or SDTM & settings obj generated automatically
+  #     (2) user needs to update settings in UI manually (due to standard="none" currently, could be more conditions in future)
+  status <- reactive({
+    req(data_selected())
+    req(settings_list$settings)
+    req(!standard()=="None")
+    validateSettings(data_selected(), settings_list$settings, chart="eDish")$valid
+  })
+
+  # based on selected data set & generated/selected settings obj, generate settings page.
+  # note that module is being triggered when selected dataset changes OR when settings list changes
+  #   this could cause the module to trigger twice unecessarily in some cases because the settings are generated
+  #   AFTER the data is changed.  
+  settingsUI_list <- reactiveValues()  ### initialize reactive values for the UI inputs
+
+ inputs <- reactive({
+   req(settings_list$settings)
+   input <- callModule(renderSettings, "settingsUI", data=data_selected, settings=settings_list$settings) 
+ })
+ 
+ # note this is getting triggered every time an input changes - even as the inputs fill in
```

Comment:
Agreed. One option is to add an "Update Settings" button at bottom of page to gate all changes.

Reply:
Decided to put this one off until after v0.4.0. Might be good to discuss details at an upcoming meeting.